builtin/providers/aws/resource_aws_instance.go

```diff
@@ -202,49 +203,57 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 		associatePublicIPAddress = v.(bool)
```

Comment:
Things like this need to be converted to the right `aws.Type`:

- https://github.com/awslabs/aws-sdk-go/blob/master/aws/types.go

So they end up like this: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_autoscaling_group.go#L137-L139